static/app/components/modals/inviteMembersModal/inviteRowControlNew.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import type {ComponentProps} from 'react';
+import {TeamFixture} from 'sentry-fixture/team';
+
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import InviteRowControlNew from 'sentry/components/modals/inviteMembersModal/inviteRowControlNew';
+import TeamStore from 'sentry/stores/teamStore';
+
+describe('InviteRowControlNew', function () {
+  const mockOnChangeEmails = jest.fn();
+  const mockOnChangeRole = jest.fn();
+  const mockOnChangeTeams = jest.fn();
+
+  const teamData = [
+    {
+      id: '1',
+      slug: 'moo-deng',
+      name: "Moo Deng's Team",
+    },
+    {
+      id: '2',
+      slug: 'moo-waan',
+      name: "Moo Waan's Team",
+    },
+  ];
+  const teams = teamData.map(data => TeamFixture(data));
+
+  const rowControlProps: ComponentProps<typeof InviteRowControlNew> = {
+    disabled: false,
+    emails: [],
+    inviteStatus: {},
+    onChangeEmails: mockOnChangeEmails,
+    onChangeRole: mockOnChangeRole,
+    onChangeTeams: mockOnChangeTeams,
+    onRemove: () => {},
+    role: '',
+    roleDisabledUnallowed: false,
+    roleOptions: [
+      {
+        id: 'member',
+        name: 'Member',
+        desc: '...',
+        minimumTeamRole: 'contributor',
+        isTeamRolesAllowed: true,
+      },
+      {
+        id: 'billing',
+        name: 'Billing',
+        desc: '...',
+        minimumTeamRole: 'contributor',
+        isTeamRolesAllowed: false,
+      },
+    ],
+    teams: [],
+  };
+
+  beforeEach(function () {
+    TeamStore.loadInitialData(teams);
+  });
+
+  it('renders', function () {
+    render(<InviteRowControlNew {...rowControlProps} />);
+
+    expect(screen.getByText('Email addresses')).toBeInTheDocument();
+    expect(screen.getByText('Role')).toBeInTheDocument();
+    expect(screen.getByText('Add to team')).toBeInTheDocument();
+  });
+
+  it('updates email addresses when new emails are inputted', async function () {
+    render(<InviteRowControlNew {...rowControlProps} />);
+    const emailInput = screen.getByLabelText('Email Addresses');
+    await userEvent.type(emailInput, 'test-space@example.com ');
+    await userEvent.type(emailInput, 'test-comma@example.com,');
+    await userEvent.type(emailInput, 'test-newline@example.com{enter}');
+    await userEvent.type(emailInput, 'test-unfocus@example.com');
+    await userEvent.tab();
+    expect(mockOnChangeEmails).toHaveBeenCalledTimes(4);
+  });
```

Comment:
In the future this test could be a bit hard to debug why it broke. the reason for that is because we are testing multiple cases in 1 expectation. (for example, if the `{enter}` stopped working for some reason, and we had to debug why - this test would fail. but we wouldn't know which case caused the error - `,` `{enter}`, `tab` and so on).

So a better way to test this would be to have something reaallly verbose like these:
```
 it('updates email addresses when new emails are inputted with a space', async function () {
    render(<InviteRowControlNew {...rowControlProps} />);
    const emailInput = screen.getByLabelText('Email Addresses');
    await userEvent.type(emailInput, 'test-space@example.com ');
    expect(mockOnChangeEmails).toHaveBeenCalledTimes(1);
    
    await userEvent.type(emailInput, 'test-space@example.com ');
    expect(mockOnChangeEmails).toHaveBeenCalledTimes(2);
  });
  
   it('updates email addresses when new emails are inputted with a enter', async function () {
    render(<InviteRowControlNew {...rowControlProps} />);
    const emailInput = screen.getByLabelText('Email Addresses');
    await userEvent.type(emailInput, 'test-space@example.com{enter}');
    expect(mockOnChangeEmails).toHaveBeenCalledTimes(1);
    
    await userEvent.type(emailInput, 'test-space@example.com{enter}');
    expect(mockOnChangeEmails).toHaveBeenCalledTimes(2);
  });
```

But that's a lot to maintain, especially for all the cases to cover. so jest introduced [`each`](https://jestjs.io/docs/api#describeeachtablename-fn-timeout) which helps write these very repetitive tests. 
```js
describe('email input tests', () => {
  describe.each([
    {email: 'test-space@example.com', delimiter: ' '},
    {email: 'test-comma@example.com', delimiter: ','},
    {email: 'test-newline@example.com', delimiter: '{enter}'},
    /* note the tab case doesn't work as it's a special case from the others */
  ])('updates email addresses when new emails are inputted with '$delimiter', ({email, delimiter}) => {
    it(`invokes the mock correctly with one`, () => {
      const emailInput = screen.getByLabelText('Email Addresses');
      await userEvent.type(emailInput, `${email}${delimiter}`);
      expect(mockOnChangeEmails).toHaveBeenCalledTimes(1);
    });
  
    it(`invokes the mock correctly with many`, () => {
      const emailInput = screen.getByLabelText('Email Addresses');
      await userEvent.type(emailInput, `${email}${delimiter}`);
      await userEvent.type(emailInput, `${email}${delimiter}`);
      await userEvent.type(emailInput, `${email}${delimiter}`);
      expect(mockOnChangeEmails).toHaveBeenCalledTimes(3);
    });
  });
});
```
This final way allows you to have the coverage of all of those other tests being written out, but the simplicity of writing it once. (Ironically, i think you'll still want to keep something like the original case to ensure a mix of cases can happen... for that you could even build out the above example more to include features like `tab` etc and built into the order of an array of delimiters.